### DISCUSSION TITLE
Testing the addon

### DISCUSSION THREAD

**Discussion by Valinakova** [Hoimar/Planet-Generator#7] [Show and tell]
Hi again. I decide to try creating original worlds using this addon. This was an attempt to emulate a planet sized version of IO moon.
![image](https://user-images.githubusercontent.com/81381626/114225614-63ad2d00-9927-11eb-8979-8e8ad18c6839.png)
Frame rates seem a bit low, when I fly through very fast, the frame rates go even lower.
That might be because I'm using RTX 550 which isn't powerful. I believe it's less than 1/10th the performance of GTX 3080.  
What is your experience with frame rates? What graphics card are you using.

I customized the atmosphere shader by making it much thinner in elevation, however, it's much more opaque. I increased light density to 30 instead of living it at 1.
![image](https://user-images.githubusercontent.com/81381626/114226196-21d0b680-9928-11eb-8504-621876c501e1.png)

New gradient I'm playing with:
![image](https://user-images.githubusercontent.com/81381626/114226301-4593fc80-9928-11eb-91eb-032dcc7a6393.png)

I reduced the "strength" so mountains relative to radius of planet would be smaller.
![image](https://user-images.githubusercontent.com/81381626/114228816-dae4c000-992b-11eb-9de6-83f688db73c6.png)

Sometimes, the program will say resource file not found when it's clearly there. I close the program and restart and it worked, without me changing anything. There appears to be other errors that appear. Are you getting similar warnings?
![image](https://user-images.githubusercontent.com/81381626/114227984-9a387700-992a-11eb-995e-1fa7ce0f99b5.png)

This below is a project I was working on blender. My attempt at creating a procedurally realistic planet. I will be attempting to recreate this with the godot visual shaders. Godot's visual nodes are much more difficult to work with compared to blender, but maybe there is a way.
![image](https://user-images.githubusercontent.com/81381626/114227455-e0410b00-9929-11eb-9d04-ad80707af876.png)

To get textures like this, I believe definitely the ability to mix noise and distort them will be essential.

**Comment by Hoimar** [comment]
This looks awesome! Really great to see my addon is useful for you.

Yes, I sometimes get these error messages and warnings too. I'm currently reworking the internals a bit to make it more stable and resource efficient so I don't leave orphaned objects and nodes left and right :smile:.

Regarding framerate: It can be both your CPU or GPU bottlenecking the framerate. If you fly very fast at low height, a lot of terrain patches will be generated simultaneously, which is quite inefficient, I'm working on solving that, so performance should definitely improve there. Your GPU may also be stressed, you can try to lower the mesh resolution in planet settings a bit, 80 is quite high. The atmosphere shader is also very taxing, you can reduce the Steps I and Steps L values which will trade quality for performance.

The atmosphere shader is kind of limited to small dimensions and hard to configure, it will need to be replaced to create large worlds (say, a planet with a radius of 6000 like Earth with 12000 km diameter).

**Edit:**
I also didn't test with many planets in the scene, that may decrease performance a bit too. You can also check for a few things in the Godot Debugger: Under Monitors, how many Nodes, Objects, Draw Calls, FPS etc. Under Profiler, which script function eats the most time etc.
Looking at your third screenshot, you may want to increase your Z-far plane in View > Settings, it already looks like part of the scene is cut off by the far clip of the editor camera. In the next Godot version, zooming will additionally be limited to your near/far plane settings in the editor.
